modules/python/python-logmsg.c

```diff
@@ -158,6 +158,29 @@ py_log_message_new(LogMessage *msg)
   return (PyObject *) self;
 }
 
+static PyObject *
+py_log_message_new_empty(PyTypeObject *subtype, PyObject *args, PyObject *kwds)
+{
+  const gchar *message = NULL;
+  gint message_length = 0;
+
+  static const gchar *kwlist[] = {"message", NULL};
+  if (!PyArg_ParseTupleAndKeywords(args, kwds, "|z#", (gchar **) kwlist, &message, &message_length))
+    return NULL;
+
+  PyLogMessage *self = (PyLogMessage *) subtype->tp_alloc(subtype, 0);
+  if (!self)
+    return NULL;
+
+  self->msg = log_msg_new_empty();
+  invalidate_cached_time();
```

Comment:
This was a point where I had enough :) The invalidation is here, because it is not called automatically, since it depends on ivykis. Threaded sources are not ivykis based, but I had to initialize ivykis, because all source threads require ivykis initialization because of the way we implemented the mark_freq timer...

TLDR: the invalidation is here because of mark messages :D  

We should definitely fix this whole thing.